packages/core/src/access.ts

```diff
@@ -29,27 +29,29 @@ export const projectAdminResourceTypes = [
  *
  * Codes taken from http://hl7.org/fhir/codesystem-restful-interaction.html
  */
-export enum AccessPolicyInteraction {
-  READ = 'read',
-  VREAD = 'vread',
-  UPDATE = 'update',
-  PATCH = 'patch',
-  DELETE = 'delete',
-  HISTORY = 'history',
-  HISTORY_INSTANCE = 'history-instance',
-  HISTORY_TYPE = 'history-type',
-  HISTORY_SYSTEM = 'history-system',
-  CREATE = 'create',
-  SEARCH = 'search',
-  SEARCH_TYPE = 'search-type',
-  SEARCH_SYSTEM = 'search-system',
-  SEARCH_COMPARTMENT = 'search-compartment',
-  CAPABILITIES = 'capabilities',
-  TRANSACTION = 'transaction',
-  BATCH = 'batch',
-  OPERATION = 'operation',
-}
-const resourceReadInteractions = [
+export const AccessPolicyInteraction = {
+  READ: 'read',
+  VREAD: 'vread',
+  UPDATE: 'update',
+  PATCH: 'patch',
+  DELETE: 'delete',
+  HISTORY: 'history',
+  HISTORY_INSTANCE: 'history-instance',
+  HISTORY_TYPE: 'history-type',
+  HISTORY_SYSTEM: 'history-system',
+  CREATE: 'create',
+  SEARCH: 'search',
+  SEARCH_TYPE: 'search-type',
+  SEARCH_SYSTEM: 'search-system',
+  SEARCH_COMPARTMENT: 'search-compartment',
+  CAPABILITIES: 'capabilities',
+  TRANSACTION: 'transaction',
+  BATCH: 'batch',
+  OPERATION: 'operation',
+} as const;
+export type AccessPolicyInteraction = (typeof AccessPolicyInteraction)[keyof typeof AccessPolicyInteraction];
```

Comment:
Just curious, how does this work, having two exports with the same name but one is a type?

Reply:
See "Alternative 2: Object Literals" here: https://maxheiber.medium.com/alternatives-to-typescript-enums-50e4c16600b1

In TypeScript, values and types exist in two separate "namespaces" or scopes:

1. **Value space**: Contains variables, functions, classes, enums, and other entities that exist at runtime in JavaScript.

2. **Type space**: Contains interfaces, type aliases, and other type constructs that exist only during compilation and are erased in the JavaScript output.

TypeScript determines from the context whether you're referring to the value or the type:

```typescript
// Using the value
const action = AccessPolicyInteraction.READ;

// Using the type
function doSomething(interaction: AccessPolicyInteraction) { ... }
```